### PR TITLE
create inotify fd with close-on-exec + deflake inotify stress test

### PR DESCRIPTION
**Pull request by pattyshack**
(on behalf of @msolo) inotify fd should not leak to children processes (this matches the kqueue implementation behavior).

### CODE REVIEW

**Comment by msolo** [comment]
Thanks for submitting this. Been on my list of things to push back for over a year.

**Comment by pattyshack** [comment]
Looks like the project's travis config is broken (the linter doesn't build in go 1.5).  Otherwise, the change seems fine

**Comment by nathany** [comment]
I merged #175. Want to rebase and give this another go on Travis?

File: inotify.go [hunk 1]
```diff
@@ -36,7 +37,7 @@ type Watcher struct {
 // NewWatcher establishes a new watcher with the underlying OS and begins waiting for events.
 func NewWatcher() (*Watcher, error) {
 	// Create inotify fd
-	fd, errno := unix.InotifyInit()
+	fd, errno := unix.InotifyInit1(syscall.IN_CLOEXEC)
```
Comment (nathany) [inline inotify.go]: We need to be careful with this change, possibly testing it manually as Travis CI doesn't give us coverage of `arm` and `ppc64` architectures that fsnotify currently supports.

File: inotify_test.go [hunk 2]
```diff
@@ -157,92 +156,105 @@ func TestInotifyCloseCreate(t *testing.T) {
...
-				if err != nil {
-					t.Fatalf("Signal failed: %v", err)
-				}
-			case <-killchan:
-				return
-			}
-		}
-	}()
+	doneChan := make(chan struct{})
+	errChan := make(chan error, 1000)
```
Comment (nathany) [inline inotify_test.go]: does this channel need to be buffered?

File: inotify_test.go [hunk 3]
```diff
@@ -157,92 +156,105 @@ func TestInotifyCloseCreate(t *testing.T) {
...
-	}()
+			case <-after:
+				for ; i > 0; i-- {
+					testFile := fmt.Sprintf("%s%d", testFilePrefix, i)
+					err = os.Remove(testFile)
+					if err != nil {
+						errChan <- fmt.Errorf("Remove failed: %v", err)
+					}
+
+					time.Sleep(time.Millisecond)
```
Comment (nathany) [inline inotify_test.go]: I'm curious why the 1ms sleep is needed.

File: inotify.go (hunk 1 above)
Comment (nathany) [inline inotify.go]: Hoping the Go team will help us out with that, but I'm not sure when they'll have time. https://github.com/golang/go/issues/17312

File: inotify_test.go (hunk 3 above)
Comment (nathany) [inline inotify_test.go]: This stress test is pretty lengthy, so it might be nice to add a few more comments for future maintainers.

**Comment by pattyshack** [comment]
(i totally messed up my branch.  will send another pull request)

**Comment by nathany** [comment]
okay

**Pull request closed without merging**